packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -463,6 +470,13 @@ class _PageViewState extends State<PageView> {
   @override
   Widget build(BuildContext context) {
     final AxisDirection axisDirection = _getDirection(context);
+    ScrollPhysics physics = widget.physics;
+    if (physics == null) {
+      physics = widget.pageSnapping ? _kPagePhysics : _kNonSnappingPhysics;
+    } else if (widget.pageSnapping) {
+      physics = _kPagePhysics.applyTo(physics);
+    }
```

Comment:
in the non-snapping, no-custom-physics case, i think you can just pass null rather than having to have _kNonSnappingPhysics.

Reply:
Updated. I also discovered that _kPagePhysics.applyTo(null) is valid, so the line can be further simplified and remove the null check.